src/test/scala/github/gphat/censorinus/ClientSpec.scala

```diff
@@ -0,0 +1,47 @@
+package github.gphat.censorinus
+
+import org.scalatest._
+import scala.collection.mutable.ArrayBuffer
+import github.gphat.censorinus.statsd.Encoder
+
+class TestSender extends MetricSender {
+  val buffer = new ArrayBuffer[String]()
+
+  def awaitMessages(n: Int): List[String] = {
+    var i = 0
+    while (i < 100 && buffer.size < n) {
+      Thread.sleep(10)
```

Comment:
I think ScalaTest actually has a thing to do this automagically and Iw as being a bit lazy. Like, `eventually` or something?